rosbag2_storage/include/rosbag2_storage/yaml.hpp

```diff
@@ -41,6 +42,27 @@ void optional_assign(const Node & node, std::string field, T & assign_to)
   }
 }
 
+template<>
+struct convert<std::unordered_map<std::string, std::string>>
+{
+  static Node encode(const std::unordered_map<std::string, std::string> & custom_data)
+  {
+    Node node;
+    for (const auto & it : custom_data) {
+      node[it.first] = it.second;
+    }
+    return node;
+  }
+
+  static bool decode(const Node & node, std::unordered_map<std::string, std::string> & custom_data)
+  {
+    for (YAML::const_iterator it = node.begin(); it != node.end(); ++it) {
+      custom_data.emplace(it->first.as<std::string>(), it->second.as<std::string>());
+    }
+    return true;
+  }
+};
+
```

Comment:
I think it would make sense to put this `convert` specialization to the `metadata_io.cpp` as others `convert` template specializations related to the metadata.

Reply:
Both `metadata_io.cpp` and `storage_options.cpp` needs this conversion so I placed it here

Reply:
@t0ny-peng I don't really like that similar definitions spread out across the different files.  
It looks like that other `yaml convert` operators was incorrectly put in the `metadata_io.cpp` file from the very beginning.  
Since we have a dedicated file for `YAML` specific operations and definitions aka `yaml.hpp` it would more appropriate to all specific yaml conversions to this file.

Could you please move other `yaml` converter operators from  `metadata_io.cpp` to the `yaml.hpp` file?
i.e. everything under the `namespace YAML`.

I have tried it locally on my machine and it's compiles and passes unit tests.

Reply:
@MichaelOrlov OK moved here